### PR TITLE
encode testdata containing potentially malicious code with base64

### PR DESCRIPTION
**Pull request by b-maldoca**
… perform required code changes

### CODE REVIEW

File: maldoca/base/file.cc [hunk 1]
```diff
@@ -223,6 +229,16 @@ absl::Status GetContents(const std::string& path, std::string* contents) {
     }
     absl::StrAppend(contents, absl::string_view(buf, ret));
   }
+  std::string content_tmp = *contents;
```
Comment (dtao-oss) [inline maldoca/base/file.cc:232]: GetContent is a general utility and this introduces unexpected behavior for general use. I think you should add a parameter "decode_as_base64" with default false so the caller controls if decoding is needed.  You can move this naming logic to a wrapper function in maldoca/maldoca/base/testing/test_utils.h to set decode_as_base64 based on names just for testing.

File: maldoca/base/file.cc [hunk 2]
```diff
@@ -223,6 +229,16 @@ absl::Status GetContents(const std::string& path, std::string* contents) {
...
+  // base64 decode file if required. However, ignore textproto files.
+  if (path.find(kFileNameBase64EncodingIndicator) != std::string::npos &&
```
Comment (dtao-oss) [inline maldoca/base/file.cc:234]: you're supposed to use absl::StrContains rather than find()

File: maldoca/base/file.cc (hunk 1 above)
Comment (dtao-oss) [reply maldoca/base/file.cc:232]: also you only need declare content_tmp if you need to decode so should only be done in the if(decode) clause.  Then use std::move to move the data not copy.

**Review by dtao-oss — CHANGES_REQUESTED** [review]

File: maldoca/base/file.cc (hunk 2 above)
Comment (b-maldoca) [reply maldoca/base/file.cc:234]: done

File: maldoca/base/file.cc (hunk 1 above)
Comment (b-maldoca) [reply maldoca/base/file.cc:232]: added utility function in test_utils and a default decode_as_base64 to GetContents.

**Review by dtao-oss — APPROVED** [review]

**Pull request merged**